context/src/main/java/io/grpc/Deadline.java

```diff
@@ -164,13 +164,19 @@ public int compareTo(Deadline that) {
     return 0;
   }
 
-  /** Time source representing nanoseconds since fixed but arbitrary point in time. */
-  abstract static class Ticker {
+  /**
+   * Time source representing nanoseconds since fixed but arbitrary point in time.
+   *
+   * <p>This is public for testing purposes.
+   */
+  public interface DeadlineTicker {
```

Comment:
Because I don't want users to stutter when using the Guava ticker

`class MyTicker extends Ticker implements io.grpc.Ticker {}`

Naming it DeadlineTicker limits scope, and prevent people from autocompleting the wrong one.